docs/user-guide/migrate-to-greptimedb/migrate-from-clickhouse.md

```diff
@@ -0,0 +1,237 @@
+ trace_id STRING SKIPPING INDEX,
+ span_id STRING SKIPPING INDEX,
+ parent_span_id STRING SKIPPING INDEX,
+ duration DOUBLE,
+ tags STRING,    -- If this is structured JSON, either store it as-is or use the pipeline to parse fields
+ PRIMARY KEY (service, operation),
+ TIME INDEX (`timestamp`)
+ );
+```
+
+**Notes:**
+- `service` and `operation` serve as tags, supporting trace scheduling and aggregate queries by service or operations.
+- `trace_id`, `span_id`, and `parent_span_id` use skip indexes but are not part of the primary key.
+- High-cardinality fields are set as fields for efficient writes. For complex properties like `tags`, JSON storage is recommended, and they can be expanded using GreptimeDB’s ETL - [Pipeline](/user-guide/logs/quick-start.md#write-logs-by-pipeline) if necessary.
+- Depending on overall business volume, consider whether to partition traces into multiple tables (such as in massive multi-service environments).
+
+---
+
+## Dual-write Strategy for Safe Migration
+
+During the migration process, to avoid data loss or inconsistent writes, adopt a dual-write approach:
+- The application should write to both ClickHouse and GreptimeDB simultaneously, running the two systems in parallel.
+- Validate and compare data using logs and checks to ensure data consistency. Once the data has been fully validated, you can switch fully over.
+
+---
+
+## Exporting and Importing Historical Data
+
+1. **Enable dual-write before migration**
+The application should write to both ClickHouse and GreptimeDB. Check for data consistency to reduce the risk of missing data.
+
+2. **Data export from ClickHouse**
+Use ClickHouse’s native command to export data as CSV, TSV, Parquet, or other formats. For example:
+```sh
+clickhouse client --query="SELECT * FROM example INTO OUTFILE 'example.csv' FORMAT CSVWithNames"
+```
+The exported CSV will look like:
+```csv
+"timestamp","host","app","metric","value"
+"2024-04-25 10:00:00","host01","nginx","cpu_usage",12.7
+"2024-04-25 10:00:00","host02","redis","cpu_usage",8.4
+"2024-04-25 10:00:00","host03","postgres","cpu_usage",15.3
+"2024-04-25 10:01:00","host01","nginx","cpu_usage",12.5
+"2024-04-25 10:01:00","host01","nginx","mem_usage",1034.5
+"2024-04-25 10:01:00","host02","redis","mem_usage",876.2
+"2024-04-25 10:01:00","host03","postgres","mem_usage",1145.2
+"2024-04-25 10:02:00","host01","nginx","disk_io",120.3
+"2024-04-25 10:02:00","host02","redis","disk_io",95.1
+"2024-04-25 10:02:00","host03","postgres","disk_io",134.7
+"2024-04-25 10:03:00","host02","redis","mem_usage",874
+"2024-04-25 10:04:00","host03","postgres","cpu_usage",15.1
+```
+
+3. **Data import into GreptimeDB**
+GreptimeDB currently supports batch data import via SQL commands or [REST API](/reference/http-endpoints.md#protocol-endpoints). For large datasets, import in batches.
+Use the [`COPY FROM` command](/reference/sql/copy.md#copy-from) to import:
+```sql
+  COPY example FROM "/path/to/example.csv" WITH (FORMAT = 'CSV');
+```
```

Comment:
We need to create table and specify primary keys first.

Reply:
Yes, it's already presented previous. But I'll add a note.